src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -471,27 +495,58 @@ class Simplex_tree {
     auto root_source = complex_source.root_;
 
     // root members copy
-    root_.members() = Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
+    root_.members() =
+        Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
     // Needs to reassign children
     for (auto& map_el : root_.members()) {
       map_el.second.assign_children(&root_);
     }
-    rec_copy(&root_, &root_source);
+    rec_copy<Options::store_key>(
+        &root_, &root_source, [](const Filtration_value& fil) -> const Filtration_value& { return fil; });
+  }
+
+  // Copy from complex_source to "this"
+  template<typename OtherSimplexTreeOptions, typename F>
+  void copy_from(const Simplex_tree<OtherSimplexTreeOptions>& complex_source, F&& translate_filtration_value) {
+    null_vertex_ = complex_source.null_vertex_;
+    filtration_vect_.clear();
+    dimension_ = complex_source.dimension_;
+    auto root_source = complex_source.root_;
+
+    // root members copy
+    for (auto& p : root_source.members()){
+      auto it = root_.members().try_emplace(root_.members().end(), p.first);
+      it->second.assign_children(&root_);
+      it->second.assign_filtration(translate_filtration_value(p.second.filtration()));
+      if constexpr (Options::store_key && OtherSimplexTreeOptions::store_key) it->second.assign_key(p.second.key());
+    }
+
+    rec_copy<OtherSimplexTreeOptions::store_key>(&root_, &root_source, translate_filtration_value);
```

Comment:
Don't we want something like
```suggestion
    rec_copy<OtherSimplexTreeOptions::store_key && Options::store_key>(&root_, &root_source, translate_filtration_value);
```
? Or is that already tested elsewhere?

Reply:
I only care if the source complex stores the key or not. If the target stores keys but not the source, the `Node` constructor will automatically store -1 as key.